eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationListConverter.php

```diff
@@ -217,6 +227,13 @@ public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefin
             $fieldSettings['selectionMethod'] = (int)$selectionType->getAttribute('value');
         }
 
+        if (
+            ($selectionType = $dom->getElementsByTagName('selection_limit')->item(0)) &&
+            $selectionType->hasAttribute('value')
+        ) {
+            $fieldSettings['selectionLimit'] = (int)$selectionType->getAttribute('value');
```

Comment:
Copy & paste typo: `$selectionLimit`, you're overwriting variable used in the previous block.